one-sdk/src/main/java/tv/accedo/one/sdk/implementation/IfModifiedTask.java

```diff
@@ -12,104 +12,137 @@
 
     private AccedoOneImpl accedoOneImpl;
     private Context context;
     private String url;
-    
-    public IfModifiedTask(AccedoOneImpl accedoOneImpl, Context context, String url){
+
+    public IfModifiedTask(AccedoOneImpl accedoOneImpl, Context context, String url) {
         this.accedoOneImpl = accedoOneImpl;
         this.context = context;
         this.url = url;
     }
-    
+
     @SuppressWarnings("unchecked")
     public <O> O run(ThrowingParser<byte[], O, AccedoOneException> parser) throws AccedoOneException {
         cleanup(context, false);
 
         Request request = accedoOneImpl.createRestClient(url);
+        String cacheKey = null;
+        String timestampCacheKey = null;
+        String profileId = null;
 
-        String key = getCacheKey(request.getUrl(), accedoOneImpl.getAppKey(), accedoOneImpl.getGid());
-        String timestampKey = getTimestampCacheKey(request.getUrl(), accedoOneImpl.getAppKey(), accedoOneImpl.getGid());
-
-        //Try to fetch a response.. Remember, even createSessionedRestClient() might fail if we are offline
+        //Try to fetch a response.. Remember, even getSession() might fail if we are offline
         Throwable caughtException = null;
         Response response = null;
-        try{
+        try {
+            // Get session, profile, and the cachekeys for the given profile
+            String sessionKey = accedoOneImpl.getSession();
+            profileId = accedoOneImpl.getCurrentSession().getProfile().getProfileId();
+            cacheKey = getCacheKey(request.getUrl(), accedoOneImpl.getAppKey(), profileId);
+            timestampCacheKey = getTimestampCacheKey(request.getUrl(), accedoOneImpl.getAppKey(), profileId);
+
             //If-Modified-Since
-            if(InternalStorage.exists(context, timestampKey)){
-                Long lastFetchedAt = (Long) InternalStorage.read(context, timestampKey);
-                if(lastFetchedAt!=null){
+            if (InternalStorage.exists(context, timestampCacheKey)) {
+                Long lastFetchedAt = (Long) InternalStorage.read(context, timestampCacheKey);
+                if (lastFetchedAt != null) {
                     request.addHeader("If-Modified-Since", sdfIfModifiedSince.format(new Date(lastFetchedAt)));
                 }
             }
-            
+
             //Connect
-            response = request.addHeader(Constants.HEADER_SESSION, accedoOneImpl.getSession()).connect(new AccedoOneResponseChecker());
-            
-        }catch(AccedoOneException e){
+            response = request.addHeader(Constants.HEADER_SESSION, sessionKey).connect(new AccedoOneResponseChecker());
+
+        } catch (Exception e) {
             Utils.log(e);
-            Utils.log(Log.INFO, "Something went wrong. Going into offline mode for: "+ request.getUrl());
+            Utils.log(Log.INFO, "Something went wrong. Going into offline mode for: " + request.getUrl());
             caughtException = e.getCause();
```

Comment:
If we catch Exception not AccedoOneException, this should probably be e, not e.getCause() ...